ansible/files/bootstrap/telemetry.sh

```diff
@@ -209,6 +221,35 @@ rm RRO-3.2.1-el6.x86_64.tar.gz
 cd RRO-3.2.1; sudo ./install.sh; cd $HOME
 $ANACONDA_PATH/bin/pip install rpy2
 
+# Setup metrics reporting
+if [ "$METRICS_PROVIDER" = "datadog" ] ; then
+    # Pull down and decrypt Datadog API key; see https://github.com/mozilla/emr-bootstrap-spark/pull/485
+    DD_DIR=$(mktemp -d)
+    aws s3 cp $TELEMETRY_CONF_BUCKET/packages/sops-3.0.4-1.x86_64.rpm $DD_DIR/
+    aws s3 cp $TELEMETRY_CONF_BUCKET/credentials/emr.yaml $DD_DIR/
+    sudo yum -y localinstall $DD_DIR/sops-3.0.4-1.x86_64.rpm
+    export DD_API_KEY=$(/usr/local/bin/sops --decrypt --extract '["datadog_agent::api_key"]' $DD_DIR/emr.yaml)
+    # Install datadog-agent; see https://app.datadoghq.com/account/settings#agent/aws
+    bash -c "$(curl -sLS https://raw.githubusercontent.com/DataDog/datadog-agent/master/cmd/agent/install_script.sh)"
+    rm -r "$DD_DIR"
+
+    # We opt not use to use https://docs.datadoghq.com/integrations/spark/
+    # Instead, we standardize on statsd metrics using Spark's StatsDSink
+    # and configure tags in the agent that will be applied by dogstatsd.
+    # This should make switching to a new metrics provider relatively straightforward.
+    sudo tee -a /etc/datadog-agent/datadog.yaml <<EOF
+
+tags:
+- cluster_name:${CLUSTER_NAME}
+- cluster_id:${CLUSTER_ID}
+$(describe_cluster '.Cluster.Tags[] | "- \(.Key):\(.Value)"')
+EOF
+
+    sudo restart datadog-agent
+elif [ -n "$METRICS_PROVIDER" ]; then
```

Comment:
should this just be else? In case the user submits empty string? Or should we use a different flag altogether e.g. --use-datadog-metrics so we don't have to arg check

Reply:
This was an attempt to stick with existing convention for other options. For `--email` and `--efs-dns`, if the options aren't supplied, then we never set the corresponding variable and we use a `-n "$EMAIL"` check to see if we need to do anything.

I'm not too concerned about the possibility of a user accidentally passing empty string as the argument. It seems more likely that they'd pass it with no argument, in which case `shift` would fail or the next flag would get swallowed.